app/src/main/java/nu/yona/app/ui/dashboard/PerDayFragment.java

```diff
@@ -250,74 +227,91 @@ else if (embeddedYonaActivity != null && embeddedYonaActivity.getLinks() != null
 	private void getDayActivity(boolean loadMore)
 	{
 		final EmbeddedYonaActivity embeddedYonaActivity = YonaApplication.getEventChangeManager().getDataState().getEmbeddedDayActivity();
-		if ((embeddedYonaActivity == null || embeddedYonaActivity.getPage() == null)
-				|| (embeddedYonaActivity != null && embeddedYonaActivity.getPage() != null && embeddedYonaActivity.getPage().getNumber() < embeddedYonaActivity.getPage().getTotalPages()))
+		if (embeddedYonaActivity == null || embeddedYonaActivity.getPage() == null || embeddedYonaActivity.getPage() != null && embeddedYonaActivity.getPage().getNumber() < embeddedYonaActivity.getPage().getTotalPages())
```

Comment:
Some day (not now), we need to refactor this too. It's a brilliant piece of JavaScript, but this is Java and we need to know what we're doing and getting.